core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4187,6 +4188,12 @@ public void doRestart(StaplerRequest req, StaplerResponse rsp) throws IOExceptio
             return;
         }
 
+        if (isRestClientAndPost(req)) {
+            rsp.setStatus(HttpURLConnection.HTTP_OK);
```

Comment:
should this not be `204` I have not seen anything setting any content.  it differs with the safe restart too which uses `202`